jetty-ee10/jetty-ee10-proxy/src/main/java/org/eclipse/jetty/ee10/proxy/AsyncMiddleManServlet.java

```diff
@@ -436,10 +437,20 @@ public void onHeaders(Response serverResponse)
         }
 
         @Override
-        public void onContent(final Response serverResponse, ByteBuffer content, final Callback callback)
+        public void onContent(Response serverResponse, Content.Chunk chunk, Runnable demander)
         {
             try
             {
+                Callback callback = Callback.from(() ->
+                {
+                    chunk.release();
+                    demander.run();
+                }, (x) ->
+                {
+                    chunk.release();
+                    serverResponse.abort(x);
+                });
```

Comment:
I'd move this callback outside the `try`, so it can be used in the `catch` below.

Also, using `Callback.from(chunk::release, Callback.from(demander::run, serverResponse::abort));` seems shorter?